### DISCUSSION TITLE
How can I get the list of available country?

### DISCUSSION THREAD

**Discussion by mingrui-liu** [vacanza/holidays#738] [Q&A] [answered]
When I am using the holiday package, sometimes I faced the error that the country is not supported. How can I get the list of the available country so that I can handle this exception?

**Comment by arkid15r** [comment, marked as answer]
You can use https://python-holidays.readthedocs.io/en/latest/api.html#holidays.utils.list_supported_countries to get the supported countries list.

**Reply by fcamuz** [reply]
Hi, I see China is among the available countries on some other documentation but the library does not have holidays for China. Is there a way to get China holidays as well?

**Reply by dr-prodigy** [reply]
China is in fact supported, code sample:

```
>>> import holidays
>>> import datetime
>>> cn = holidays.CN()
>>> datetime.date(2022,10,1) in cn
True
>>> cn
{datetime.date(2022, 1, 1): "New Year's Day", datetime.date(2022, 5, 1): 'Labour Day', datetime.date(2022, 2, 1): 'Chinese New Year (Spring Festival)', datetime.date(2022, 2, 2): 'Chinese New Year (Spring Festival)', datetime.date(2022, 2, 3): 'Chinese New Year (Spring Festival)', datetime.date(2022, 10, 1): 'National Day', datetime.date(2022, 10, 2): 'National Day', datetime.date(2022, 10, 3): 'National Day', datetime.date(2022, 4, 5): 'Tomb-Sweeping Day', datetime.date(2022, 6, 3): 'Dragon Boat Festival', datetime.date(2022, 9, 10): 'Mid-Autumn Festival'}
```
